lib/semantic_puppet/version.rb

```diff
@@ -126,70 +110,65 @@ def eql?(other)
     alias == eql?
 
     def to_s
-      "#{major}.#{minor}.#{patch}" +
-      (@prerelease.nil? || prerelease.empty? ? '' : "-" + prerelease) +
-      (@build.nil?      || build.empty?      ? '' : "+" + build     )
+      s = "#{@major}.#{@minor}.#{@patch}"
+      unless @prerelease.nil? || @prerelease.empty?
+        s << '-'
+        @prerelease.each { |p| s << p.to_s << '.' }
+        s.chomp!('.')
```

Comment:
Simpler in one sense, but also slower. I think stringifying a version will be fairly common, and there for somewhat performance critical. I wanted to avoid creating the extra string that will be the result of that and I also wanted to avoid the join since it's slow. Appending to a single string is faster.